### DISCUSSION TITLE
[@types/chart.js]: @types/chart.js@2.9.37 not match with @chart.js@2.9.37

### DISCUSSION THREAD

**Discussion by s-nt-s** [DefinitelyTyped/DefinitelyTyped#63641] [Issues with a @types package]
If I install:

```
npm install charts.js@2.9.3
npm install --save-dev @types/charts.js@2.9.3
```

and use:

```
const dat:Chart.ChartConfiguration = {
           // stuff
           options: {
                 onResize: function (ch:Chart, sz:Chart.ChartSize) {
                          // stuff
                 }
           }
}
```

I get this error in `onResize`: Type '(ch: Chart, sz: ChartSize) => void'  is not assignable to type '(this: Chart, newSize: ChartSize) => void'.ts(2322)

That is because of in `@types/charts.js@2.9.3` (`src/node_modules/@types/chart.js/index.d.ts`) `onResize` is typed like `onResize` is in newer versions of `charts.js`:

```
interface ChartOptions {
    // stuff
    onResize?(this: Chart, newSize: ChartSize): void;
    // stuff
}
```

But that it wasn't its type in version 2.9.3, as you can see in https://www.chartjs.org/docs/2.9.4/general/responsive.html it said:

> `onResize`: Called when a resize occurs. Gets passed two arguments: the chart instance and the new size.

Is `@types/charts.js@2.9.3` wrong? Which version of `@types/charts.js` I need to work with `charts.js@2.9.3`?

[a comment by an automated account is not shown]

**Comment by lekoaf** [comment]
I'm not sure what's wrong here, but you could try and use the latest version of the TypeDefinitions, which is 2.9.37.
The Chart.js versioning doesn't closely follow the types versioning.

Then again, I wouldn't recommend using such an old version of Chart.js either. I think 4.0 was release the other day and since Chart.js v3 it's been rewritten with Typescript, so you don't need these typedefinitions anymore.

**Reply by s-nt-s** [reply]
I have tried with all version listed here: https://www.npmjs.com/package/@types/chart.js?activeTab=versions
and in all version that `onResize` appeared is like `onResize?(this: Chart, newSize: ChartSize): void;`

Upgrade version of  Chart.js is not possible at the moment.

**Reply by lekoaf** [reply]
Okay. I guess the next step is to do a PR that will fix your issue.
